src/proxy/pool.rs

```diff
@@ -83,6 +83,26 @@ struct ConnSpawner {
 
 // Does nothing but spawn new conns when asked
 impl ConnSpawner {
+    async fn new_unpooled_conn(
```

Comment:
Anything here we can do higher up I think, but things might change if we decide to implement pooling in this PR

Reply:
Yeah if we want double-hbone conns to be unpooled and thus need ~none of this surrounding machinery, then I'd be inclined to just start `proxy/double-hbone.rs` and use that directly.